core/Modules/MxDownload/MxDownload.php

```diff
@@ -176,6 +176,7 @@ public static function addMap(Player $player, int $mxId)
             'name' => $gbx->Name,
             'environment' => $gbx->Environment,
             'title_id' => $gbx->TitleId,
+            'folder' => substr($filename, 0, strrpos($filename, DIRECTORY_SEPARATOR)),
```

Comment:
Please use https://www.php.net/manual/en/function.dirname.php here

Reply:
This would return the whole path of the parent directory. In my opion the folder field should represent the folder the map is in relative to the Maps Directory of the server.

Reply:
I could just remove the servers map directory from the dirname if you meant that with you comment.

Reply:
```php
str_replace(Server::getMapsDirectory(), "", dirname(Server::getMapsDirectory().$map->fileName))
```

Is probably a bit easier to understand.

Reply:
I just now looked depper at the code.
Why do you try to write to the non-existing "folder" column?

Reply:
The culomn exists since you did merge the last Pull Request from me, which allows to select all Maps from the directories.

Reply:
Look at my last pull request for the changes being made.

Reply:
Oh man GitHub... I didnt want to merge that one wtf.

Reply:
k, forget my comment on the other PR.
I'll look at what you've coded and if we can use that

Reply:
What is your goal with the other PR? I thought you wanted to include directories in the matchsettings, that should be loaded aswell. What you added is some kind of directory management, why?

Reply:
Do you have discord? Maybe we could talk there about what you have imagined, I probably have another understanding of what you've planned.
My ID is Braker#6474

Reply:
Right now you had to select every single map to add to a specific Match Settings profile. My pull requests adds a tab to the Edit UI of the Matchsettings which lets you add or remove all maps from a specific folder to that profile. The same way you can remove all maps in that folder from the profile.

Reply:
Do you understand what I mean?

Reply:
Hmm, I see.
I'd rather call it add/remove all maps from directory to/from matchsettings, instead of enable/disable.

When you said you wanted to include directories I though you're going to add entries to the matchsettings like this:
`
  <map_dir>SomeSubDir</map_dir>
  <map>
    <file>MX/200_Hypa_Hypa.Map.Gbx</file>
    <ident>N15OlItvCmGbT6xr5w_hMM3Jz2g</ident>
  </map>
....`

We can keep it and continue to work on it.

Reply:
I did. As i just said it does the same how maps are currently added.